Assets/Mirror/Core/NetworkBehaviourHybrid.cs

```diff
@@ -0,0 +1,479 @@
+        [Header("Debug")]
+        public bool debugLog = false;
+
+        public virtual void Reset()
+        {
+            lastSerializedBaselineTick = 0;
+            lastDeserializedBaselineTick = 0;
+            changedSinceBaseline = false;
+        }
+
+        // user callbacks //////////////////////////////////////////////////////
+        protected abstract void OnSerializeBaseline(NetworkWriter writer);
+        protected abstract void OnDeserializeBaseline(NetworkReader reader, byte baselineTick);
+
+        protected abstract void OnSerializeDelta(NetworkWriter writer);
+        protected abstract void OnDeserializeDelta(NetworkReader reader, byte baselineTick);
+
+        // implementations must store the current baseline state when requested:
+        // - implementations can use this to compress deltas against
+        // - implementations can use this to detect changes since baseline
+        // this is called whenever a baseline was sent.
+        protected abstract void StoreState();
+
+        // implementations may compare current state to the last stored state.
+        // this way we only need to send another reliable baseline if changed since last.
+        // this is called every syncInterval, not every baseline sync interval.
+        // (see comments where this is called).
+        protected abstract bool StateChanged();
+
+        // user callback in case drops due to baseline mismatch need to be logged/visualized/debugged.
+        protected virtual void OnDrop(byte lastBaselineTick, byte baselineTick, NetworkReader reader) {}
+
+        // rpcs / cmds /////////////////////////////////////////////////////////
+        [ClientRpc(channel = Channels.Reliable)] // reliable baseline
+        void RpcServerToClientBaseline(ArraySegment<byte> data)
+        {
+            // baseline is broadcast to all clients.
+            // ignore if this object is owned by this client.
+            if (IsClientWithAuthority) return;
+
+            // host mode: baseline Rpc is also sent through host's local connection and applied.
+            // applying host's baseline as last deserialized would overwrite the owner client's data and cause jitter.
+            // in other words: never apply the rpcs in host mode.
+            if (isServer) return;
+
+            using (NetworkReaderPooled reader = NetworkReaderPool.Get(data))
+            {
+                // deserialize
+                // save last deserialized baseline tick number to compare deltas against
+                lastDeserializedBaselineTick = reader.ReadByte();
+                OnDeserializeBaseline(reader, lastDeserializedBaselineTick);
+            }
+        }
+
+        [ClientRpc(channel = Channels.Unreliable)] // unreliable delta
+        void RpcServerToClientDelta(ArraySegment<byte> data)
+        {
+            // delta is broadcast to all clients.
+            // ignore if this object is owned by this client.
+            if (IsClientWithAuthority) return;
```

Comment:
Use `includeOwner = false` or add comments why not.

Reply:
done